packages/react/src/css-map/index.ts

```diff
@@ -1,6 +1,44 @@
-import type { CSSProps, CssObject } from '../types';
+import type * as CSS from 'csstype';
+
 import { createSetupError } from '../utils/error';
 
+/**
+ * These are all the CSS props that will exist.
+ * Only 'string' and 'number' are valid CSS values.
+ *
+ * @example
+ * ```
+ * const style: CssProps = {
+ *  color: 'red',
+ *  margin: 10,
+ * };
+ * ```
+ */
+type CssProps = Readonly<CSS.Properties<string, number>>;
+
+/**
+ * Recursively typed CSS object because nested objects are allowed.
+ *
+ * @example
+ * ```
+ * const style: CssObject = {
+ *  "@media screen and (min-width: 480px)": {
+ *    ":hover": {
+ *      color: 'red'
+ *   }
+ *  }
+ * }
+ * ```
+ */
+type CssObject = Readonly<
```

Comment:
I wonder if there's a way to ensure that the user can't nest objects indefinitely, e.g. stuff like

```tsx
const styles = cssMap({
  danger: {
      veryDanger: {
        color: 'red',
      }
  },
})
```

without affecting valid use cases such as `@media` queries

Reply:
I'm not at my laptop so I can't test this out, but it does seem like there's nothing stopping the user from doing stuff like

```tsx
const styles = cssMap({
  danger: {
      veryDanger: {
        lotsofDanger: {
          evenmoreDanger: {
            spectacularDanger: {
              color: 'red',
            }
          }
        }
      }
  },
})
```

Reply:
yea, we've to allow it unless we impose stricter types.
The problem is that we don't know if the _object key_ is a selector.
In the below example, we don't know if `header` is a variant name or a selector.
```
const styles = cssMap({
  success: {
      header: { // Is it <header /> or a variant name? 🤷‍♂️
         div: {
             color: 'red'
         }
      }
  },
})
```
I thought about disallowing targeting selectors completely, but I think disallowing it is too specific to Atlassian CSS standards.

Reply:
I see, thank you for clarifying 👍

Reply:
I think we can do better here, it may be worth a play with the structure

I've made a quick spike based on vanilla extract types https://github.com/atlassian-labs/compiled/pull/new/css-map-type-spike

In https://vanilla-extract.style/documentation/styling/#media-queries, they specify some bounded keys that allow the types to be much stricter